src/test/java/uk/gov/ons/census/casesvc/messaging/SurveyLaunchedReceiverIT.java

```diff
@@ -0,0 +1,126 @@
+
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.ons.census.casesvc.service.ReceiptService.QID_RECEIPTED;
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getTestResponseManagementReceiptEvent;
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getTestResponseManagementSurveyLaunchedEvent;
+import static uk.gov.ons.census.casesvc.utility.JsonHelper.convertObjectToJson;
+
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class SurveyLaunchedReceiverIT {
+  private static final UUID TEST_CASE_ID = UUID.randomUUID();
+  private static final EasyRandom easyRandom = new EasyRandom();
+  private static final String TEST_QID = easyRandom.nextObject(String.class);
+  private static final String TEST_UAC = easyRandom.nextObject(String.class);
+
+  @Value("${queueconfig.survey-launched-queue}")
+  private String inboundQueue;
+
+  @Autowired private RabbitQueueHelper rabbitQueueHelper;
+  @Autowired private CaseRepository caseRepository;
+  @Autowired private EventRepository eventRepository;
+  @Autowired private UacQidLinkRepository uacQidLinkRepository;
+
+  @Before
+  @Transactional
+  public void setUp() {
+    rabbitQueueHelper.purgeQueue(inboundQueue);
+    eventRepository.deleteAllInBatch();
+    uacQidLinkRepository.deleteAllInBatch();
+    caseRepository.deleteAllInBatch();
+  }
+
+  @Test
+  public void testSurveyLaunchLogsEvent() throws JSONException {
+    // GIVEN
+    EasyRandom easyRandom = new EasyRandom();
+    Case caze = easyRandom.nextObject(Case.class);
+    caze.setCaseId(TEST_CASE_ID);
+    caze.setUacQidLinks(null);
+    caze.setEvents(null);
+    caze = caseRepository.saveAndFlush(caze);
+
+    UacQidLink uacQidLink = new UacQidLink();
+    uacQidLink.setId(UUID.randomUUID());
+    uacQidLink.setCaze(caze);
+    uacQidLink.setQid(TEST_QID);
+    uacQidLink.setUac(TEST_UAC);
+    uacQidLinkRepository.saveAndFlush(uacQidLink);
+
+    ResponseManagementEvent surveyLaunchedEvent = getTestResponseManagementSurveyLaunchedEvent();
+    surveyLaunchedEvent.getPayload().getReceipt().setQuestionnaireId(uacQidLink.getQid());
+
+    String json = convertObjectToJson(surveyLaunchedEvent);
+    Message message =
+        MessageBuilder.withBody(json.getBytes())
+            .setContentType(MessageProperties.CONTENT_TYPE_JSON)
+            .build();
+
+    // WHEN
+    rabbitQueueHelper.sendMessage(inboundQueue, message);
+
+    // THEN
+    List<Event> events = eventRepository.findAll();
+    assertThat(events.size()).isEqualTo(1);
+    Event event = events.get(0);
+    assertThat(event.getEventDescription()).isEqualTo("SURVEY LAUNCHED");
```

Comment:
This ALL CAPS description is not really consistent with other descriptions

Reply:
Fixed.